### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Anastasiia-Svintsova**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -1,61 +1,69 @@
...
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const children = people.filter(person => !!person.mother === true);
+  const mothers = children.map(child => child.mother);
+
+  const women = people.filter(woman => withChildren
+    ? woman.sex === 'f' && mothers.includes(woman.name)
+    : woman.sex === 'f'
+  );
```
Comment (YegorKochetkov) [inline src/averageAges.js:34]: interesting solution )

File: src/averageAges.js [hunk 2]
```diff
@@ -1,61 +1,69 @@
...
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const children = people.filter(person => !!person.mother === true);
```
Comment (YegorKochetkov) [inline src/averageAges.js:28]: what's this for `!!person.mother === true`?
maybe `person.mother !== null` enough?

**Review by YegorKochetkov — APPROVED** [review]
good)

File: src/averageAges.js (hunk 2 above)
Comment (AlinaMatuschak) [reply src/averageAges.js:28]: person.mother will be enough )
string convert to true

**Review by AlinaMatuschak — APPROVED** [review]
Good work 👌

**Pull request closed without merging**